sisl/orbital.py

```diff
@@ -1130,3 +1130,50 @@ def __setstate__(self, d):
             self.__init__(d["name"], q0=d["q0"], tag=d["tag"])
         else:
             self.__init__(d["name"], (d["r"], d["f"]), q0=d["q0"], tag=d["tag"])
+
+
+@set_module("sisl")
+class HydrogenicOrbital(AtomicOrbital):
+    r""" A hydrogen-like atomic orbital defined by an effective atomic number Z
+    in addition to the usual quantum numbers (n, l, m).
+    The returned orbital is properly normalized.
+
+    See https://en.wikipedia.org/wiki/Hydrogen-like_atom
+
+    Parameters
+    ----------
+    Zeff : float
```

Comment:
Could we perhaps just use `Z`? And emphasize `effective`? More if future orbitals also makes use of Z?